### ISSUE TITLE
Add #destroy method

### ISSUE THREAD

**Issue by tiquelou** [polarblau/smarttruncation#2]
``` html
<div id="truncateThis">Some text<div id="dynamic01></div></div>
```

truncateThis is the div on which smart truncate is being used. If you are using dynamic01 to display some dynamic data, like, say, the text of a selected item in a dropdown, nothing is rendered in dynamic01.

**Comment by polarblau** [comment]
Is it that you'd like to change the content of an element on which you've called the plugin?

**Comment by tiquelou** [comment]
correct, which is not happening. Is there something I am missing?

also, going thru the code, I realised that the plugin might not work well
with internationalization... plz do confirm.

On Tue, Mar 29, 2011 at 12:42 PM, polarblau <
reply@reply.github.com>wrote:

> Is it that you'd like to change the content of an element on which you've
> called the plugin?
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/polarblau/smarttruncation/issues/2#comment_929844

## 

_Arnab Bandyopadhyay_

_Director, Operations & New Ventures
arnab@plavaga.in | Mobile: +91 9986029040
www.plavaga.in | Jump start possibilities_

**Comment by polarblau** [comment]
You're absolutely correct what comes to the dynamic content. The plugin stores the original content of the element for you and this content would need to be updated when the content of the element is changed. However, I'm wondering if this is a logical place to add this kind of functionality.
I' could imagine adding a `destroy` method to remove the plugins functionality, giving the user the possibility to re–attach it after changing the content. Monitoring if the content has changed seems overly costly to me. What do you think?

Could you give an example on where do you see problems with the internationalization?

_I've noticed that — depending on your font–settings — you might have to quite a lot of tweaking to get things to work as you want them to. (Just as a warning ;) ) . I'm working on a test suite and some options to improve this situation as well as a solution to handle HTML entities properly._

**Comment by tiquelou** [comment]
aah! so let me try a few tweaks and I will let you know if I have any
success!! :)

**Comment by tiquelou** [comment]
Another update - I checked for the internationalization, the plugin works
just fine with non-latin scripts :)

On Tue, Mar 29, 2011 at 1:14 PM, polarblau <
reply@reply.github.com>wrote:

> You're absolutely correct what comes to the dynamic content. The plugin
> stores the original content of the element for you and this content would
> need to be updated when the content of the element is changed. However, I'm
> wondering if this is a logical place to add this kind of functionality.
> I' could imagine adding a `destroy` method to remove the plugins
> functionality, giving the user the possibility to reattach it after
> changing the content. Monitoring if the content has changed seems overly
> costly to me. What do you think?
> 
> Could you give an example on where do you see problems with the
> internationalization?
> 
> _I've noticed that  depending on your fontsettings  you might have to
> quite a lot of tweaking to get things to work as you want them to. (Just as
> a warning ;) ) . I'm working on a test suite and some options to improve
> this situation as well as a solution to handle HTML entities properly._
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/polarblau/smarttruncation/issues/2#comment_929924

## 

_Arnab Bandyopadhyay_

_Director, Operations & New Ventures
arnab@plavaga.in | Mobile: +91 9986029040
www.plavaga.in | Jump start possibilities_

**Comment by tiquelou** [comment]
Ouch! Hit a brick wall... looks like a 'destroy' method is very much
required!

On Tue, Mar 29, 2011 at 1:14 PM, polarblau <
reply@reply.github.com>wrote:

> You're absolutely correct what comes to the dynamic content. The plugin
> stores the original content of the element for you and this content would
> need to be updated when the content of the element is changed. However, I'm
> wondering if this is a logical place to add this kind of functionality.
> I' could imagine adding a `destroy` method to remove the plugins
> functionality, giving the user the possibility to reattach it after
> changing the content. Monitoring if the content has changed seems overly
> costly to me. What do you think?
> 
> Could you give an example on where do you see problems with the
> internationalization?
> 
> _I've noticed that  depending on your fontsettings  you might have to
> quite a lot of tweaking to get things to work as you want them to. (Just as
> a warning ;) ) . I'm working on a test suite and some options to improve
> this situation as well as a solution to handle HTML entities properly._
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/polarblau/smarttruncation/issues/2#comment_929924

## 

_Arnab Bandyopadhyay_

_Director, Operations & New Ventures
arnab@plavaga.in | Mobile: +91 9986029040
www.plavaga.in | Jump start possibilities_

**Comment by polarblau** [comment]
Thanks for your feedback. Good to hear that the plugin works for other languages than English as well.
What comes to the destroy method: I'll look into that as soon as I get around to. In the mean time you could use `remove()` to get rid of the wrapper and then re–insert the wrapper + content and then attach the plugin again. It's a bit of a hack but might just do the trick. Haven't tested it, though. —Let me know you get in trouble.

**Comment by tiquelou** [comment]
will do. thanks for listening!
